crates/polars-time/src/upsample.rs

```diff
@@ -130,6 +137,29 @@ fn upsample_impl(
         out.try_apply(index_column, |s| s.cast(&DataType::Date))
             .unwrap();
         Ok(out)
+    } else if matches!(s.dtype(), DataType::Int32) {
```

Comment:
check https://github.com/pola-rs/polars/pull/14669/files - could `UInt32` and `UInt64` be added here too?

Reply:
OK. I added UInt32, UInt64.